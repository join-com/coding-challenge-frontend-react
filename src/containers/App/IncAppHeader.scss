.IncAppHeader {

    margin: 1em 0 3em;
    
    img.logo {
        max-width: 100%;
    }

    > div {
        display: flex;
        flex-direction: column;
        justify-content: center;
    }
    
}
