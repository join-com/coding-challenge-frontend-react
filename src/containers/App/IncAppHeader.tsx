import * as React from 'react';

import Link from '../../components/Hyperlink/ComHyperlink';

import './IncAppHeader.scss';


const COM_NAME = 'IncAppHeader';

interface IProps {
    logo: string;
    title: string;
    subtitle: string;
}

export default (props: IProps): React.ReactElement => {
    return (
        <div className={ `${COM_NAME} row` }>
            <div className="col-3 col-xs-4 col-lg-2 pt-2 pl-3">
                <Link to="/">
                    <img className="logo" src={ props.logo } />
                </Link>
            </div>
            <div className="col-9 col-xs-8 col-lg-10">
                <h1>{ props.title }</h1>
                <h2>{ props.subtitle }</h2>
            </div>
        </div>
    );
};
