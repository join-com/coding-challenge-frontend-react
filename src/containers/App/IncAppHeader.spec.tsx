import * as React from 'react';
import { MemoryRouter } from 'react-router-dom'

import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';

import IncAppHeader from './IncAppHeader';


describe('IncAppHeader', () => {

    let node: any;
    let comp: any;

    beforeEach(() => {
        node = (
            <IncAppHeader
                logo="/test/logo.png"
                title="Lorem Ppsum"
                subtitle="dolor sit amet..."
                />
        );
        comp = mount(<MemoryRouter>{ node }</MemoryRouter>);
    });

    it('should render the component', () => {
        expect(comp.length).toBe(1);
    });

    it('should match a snapshot', () => {
        const snapshot = renderer
            .create(<MemoryRouter>{ node }</MemoryRouter>)
            .toJSON();
        expect(snapshot).toMatchSnapshot();
    });

});
