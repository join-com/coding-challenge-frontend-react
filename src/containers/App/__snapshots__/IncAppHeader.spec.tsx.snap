// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncAppHeader should match a snapshot 1`] = `
<div
  className="IncAppHeader row"
>
  <div
    className="col-3 col-xs-4 col-lg-2 pt-2 pl-3"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        className="logo"
        src="/test/logo.png"
      />
    </a>
  </div>
  <div
    className="col-9 col-xs-8 col-lg-10"
  >
    <h1>
      Lorem Ppsum
    </h1>
    <h2>
      dolor sit amet...
    </h2>
  </div>
</div>
`;
